ravenpy/utilities/gis.py

```diff
@@ -3,35 +3,146 @@
 -----------------------------------
 
 * Point coordinates are passed as shapely.geometry.Point instances.
-* BBox coordinates are passed as (lon1, lat1, lon2, lat2)
+* BBox coordinates are passed as (lon1, lat1, lon2, lat2).
 * Shapes (polygons) are passed as ?
-* All functions that require a CRS have a CRS argument with a default set to WSG84 (?)
-*
+* All functions that require a CRS have a CRS argument with a default set to WSG84.
+* GEO_URL points to the GeoSever hosting all files.
 
 """
-
 import collections
 from pathlib import Path
 from typing import Iterable, List, Sequence, Tuple, Union
 
 import fiona
 import pandas as pd
+from lxml import etree
+from owslib.fes import PropertyIsLike
+from owslib.wfs import WebFeatureService
+from requests import Request
 from shapely.geometry import Point, shape
 
 from ravenpy.utils import crs_sniffer, single_file_check
 
-GEO_URL = "http://boreas.ouranos.ca/geoserver/wfs"
+GEO_URL = "http://boreas.ouranos.ca/geoserver/"
 
 # We store the contour of different hydrobasins domains
 hybas_dir = Path(__file__).parent.parent / "data" / "hydrobasins_domains"
 hybas_pat = "hybas_lake_{}_lev01_v1c.zip"
 
 # This could be inferred from existing files in hybas_dir
 hybas_regions = ["na", "ar"]
-
 hybas_domains = {dom: hybas_dir / hybas_pat.format(dom) for dom in hybas_regions}
 
 
+def _get_location_wfs(
+    coordinates: Tuple[
+        Union[int, float, str],
+        Union[str, float, int],
+        Union[str, float, int],
+        Union[str, float, int],
+    ] = None,
+    level: int = None,
+    layer: str = True,
+) -> str:
+    """Return leveled features from a hosted data set using bounding box coordinates and WFS 1.1.0 protocol.
+
+    For geographic rasters, subsetting is based on WGS84 (Long, Lat) boundaries. If not geographic, subsetting based
+    on projected coordinate system (Easting, Northing) boundaries.
+
+    Parameters
+    ----------
+    coordinates : Tuple[Union[str, float, int], Union[str, float, int], Union[str, float, int], Union[str, float, int]]
+      Geographic coordinates of the bounding box (left, down, right, up).
+    level : int
+      Level of granularity requested for the lakes vector.
+    layer : str
+      The WFS/WMS layer name requested.
+
+    Returns
+    -------
+    str
+      A GML-encoded vector feature.
+
+    """
+    layer = f"{layer}{level}"
+
+    if coordinates is not None:
+        wfs = WebFeatureService(f"{GEO_URL}/wfs", version="1.1.0", timeout=30)
+        try:
+            resp = wfs.getfeature(
+                typename=layer, bbox=coordinates, srsname="urn:x-ogc:def:crs:EPSG:4326"
+            )
+        except Exception as e:
+            raise Exception(e)
```

Comment:
I'm aware there must be a good reason for this but what is it again?

Reply:
Not really in this instance since I don't add much here, and it could potentially hide useful exception messages. I coded this a long time ago, so it's good to be removed, I think.